frontend/src/pages/RunDetails.tsx

```diff
@@ -326,15 +326,15 @@ class RunDetails extends Page<RunDetailsProps, RunDetailsState> {
                                     <DetailsTable
                                       title='Input artifacts'
                                       fields={inputArtifacts}
-                                      valueComponent={MinioArtifactLink}
+                                      valueComponent={MinioArtifactPreview}
```

Comment:
Also here is also problematic too.
https://github.com/Bobgy/pipelines/blob/fe8928926b42a51976e2f421642924a3771afde6/frontend/src/components/DetailsTable.tsx#L106

It should return <MinioArtifactLink {...value} />, not calling the function directly.

This explains why after building an image, your UI crashes. These are unexpected behaviors for react.

Reply:
👍 i went to properly read up on react. Now I have a better idea, how jsx works.

Reply:
👍 i went to properly read up on react. Now I have a better idea, how jsx works.